### ISSUE TITLE
Unexpected behavior when deserializing to double

### ISSUE THREAD

**Issue by Another-Anonymous-User** [biojppm/rapidyaml#434]
Deserializing to double using:
```
ryml::NodeRef node = tree["key"];
double value;
node >> value;
```
The following two cases behave unexpectedly:
- When the key is followed by no value it gives a segmentation fault
- When the key is a number with a letter in a position that is not the first it returns 0 instead of giving an error.

**Comment by biojppm** [comment]
>     * When the key is followed by no value it gives a segmentation fault

TL;DR you should never call >> if you are unsure the value is empty; use `.val_is_null()` to check for this before calling, or use `.get_if()`.

You cannot deserialize a val which does not exist, ie you cannot deserialize a null val. This is [stated really early in the quickstart](https://github.com/biojppm/rapidyaml/blob/master/samples/quickstart.cpp#L513) (and see also a [sample for null values](https://github.com/biojppm/rapidyaml/blob/e432c2f390c9430a862d363c6c384f90590135ce/samples/quickstart.cpp#L2494).

As for the error,  it is actually an assertion, and as stated in the README the default error handler calls `abort()`. Are you sure it is a segfault, or is it an abort?

Also, I guess you're using an older version; in the latest release there are some important error handling fixes; see [the release notes](https://github.com/biojppm/rapidyaml/releases/tag/v0.6.0) for more info.


> 
>     * When the key is a number with a letter in a position that is not the first it returns 0 instead of giving an error.

Please provide concrete examples; the alphabetic characters matter, and their position as well.

Also, there was a fix in 0.6.0 that may be relevant for this problem; see #390 and #415.

**Comment by Another-Anonymous-User** [comment]
Thank you for the explanations and my bad I missed the part about using >> on empty values in the quick start.

> Are you sure it is a segfault, or is it an abort?
After further testing I determined its a segmentation fault when I deserialize to a double or float. Other data types tested lead to an abort.

Regarding the second issue a small correction, it does not return zero but only the numbers before the letter(s). For example:
`key: 0.r3` will return 0
`key: 34gf3` will return 34
This is the case when I deserialize to a double or float. When deserializing to an integer it leads to an expected "could not deserialize value" error.

Note: I am using the master branch of about a week ago (commit e432c2f)

**Comment by biojppm** [comment]
I've investigated this.

 * for the empty value, there was a missing check for empty string in the `read(ConstNodeRef, T*)` functions. This would cause the segfault. I've added that check.
 * the difference in behavior for `key: 34gf3` between integers and floats is indeed there. Unfortunately, this happens because the fast_float parser is doing that. If you need to handle this in your data, you can use `csubstr::first_real_span()` to ensure that all is good *before* deserializing. Something like this:
   ```cpp
   csubstr val = node.val();
   if(val.first_real_span() == val)
      node >> var;
   else
      ERROR("not a real");
   ```
